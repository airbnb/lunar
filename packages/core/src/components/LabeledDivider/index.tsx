import React from 'react';
import Text from '../Text';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

export type Props = {
  /** A label for the divider. Typically a string or i18n T node. */
  label: NonNullable<React.ReactNode>;
};

/** A horizontal divider with a label. */
export class LabeledDivider extends React.Component<Props & WithStylesProps> {
  render() {
    const { cx, styles, label } = this.props;

    return (
      <div className={cx(styles.rule)}>
        <Text small bold inline>
          {label}
        </Text>
      </div>
    );
  }
}

export default withStyles(({ ui }) => ({
  rule: {
    borderBottom: ui.border,
  },
}))(LabeledDivider);
