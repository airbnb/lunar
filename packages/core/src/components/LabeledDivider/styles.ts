import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ ui }) => ({
  rule: {
    borderBottom: ui.border,
  },
});

export default styleSheet;

export { styleSheet };
