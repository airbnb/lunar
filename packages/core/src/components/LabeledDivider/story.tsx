import React from 'react';
import LabeledDivider from '.';

export default {
  title: 'Core/LabeledDivider',
  parameters: {
    inspectComponents: [LabeledDivider],
  },
};

export function standardDividerWithLabel() {
  return <LabeledDivider label="Custom label" />;
}

standardDividerWithLabel.story = {
  name: 'Standard divider with label.',
};
