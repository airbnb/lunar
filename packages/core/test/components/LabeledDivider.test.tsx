import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import T from '../../src/components/Translate';
import LabeledDivider from '../../src/components/LabeledDivider';

describe('<LabeledDivider />', () => {
  it('renders a string as expected', () => {
    const wrapper = shallowWithStyles(<LabeledDivider label="Lorum ipsum" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a translation node as expected', () => {
    const wrapper = shallowWithStyles(
      <LabeledDivider label={<T phrase="Lorum ipsum" context="Foo" />} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
