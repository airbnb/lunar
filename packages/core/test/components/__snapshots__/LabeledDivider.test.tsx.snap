// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabeledDivider /> renders a string as expected 1`] = `
<div
  className="rule_bi7c4v"
>
  <Text
    bold={true}
    inline={true}
    small={true}
  >
    Lorum ipsum
  </Text>
</div>
`;

exports[`<LabeledDivider /> renders a translation node as expected 1`] = `
<div
  className="rule_bi7c4v"
>
  <Text
    bold={true}
    inline={true}
    small={true}
  >
    <Translate
      context="Foo"
      phrase="Lorum ipsum"
    />
  </Text>
</div>
`;
